Sources/TrimBehavior.swift

```diff
@@ -0,0 +1,86 @@
+//
+//  File.swift
+//  
+//
+//  Created by Yonas Kolb on 12/11/19.
+//
+
+import Foundation
+
+public struct TrimBehavior: Equatable {
+  var leading: Trim
+  var trailing: Trim
+
+  public enum Trim: Equatable {
+    /// nothing
+    case none
```

Comment:
I suppose we would not need this case if we would model this and trim behaviour property as optionals, this way we also won't need switch on it, use fatalError and always check if it's not `none`, instead we will have to unwrap optional which is more idiomatic.